spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/quartz/TracingJobListener.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth.instrument.quartz;
+
+import brave.Span;
+import brave.Tracer;
+import brave.Tracer.SpanInScope;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.quartz.JobListener;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * {@link org.quartz.JobListener JobListener} that will wrap a span around quartz jobs
+ * when they start and finish.
+ *
+ * @author Branden Cash
+ */
+@Component
+public class TracingJobListener implements JobListener {
+
+	static final String CONTEXT_SPAN_KEY = "TracingSpan";
+	static final String CONTEXT_SPAN_IN_SCOPE_KEY = "TracingSpanInScope";
+
+	private Tracer tracer;
+
+	@Autowired
+	public TracingJobListener(Tracer tracer) {
+		this.tracer = tracer;
+	}
+
+	@Override
+	public String getName() {
+		return getClass().getName();
+	}
+
+	@Override
+	public void jobToBeExecuted(JobExecutionContext context) {
+		String name = String.format("Job: %s; Trigger: %s",
```

Comment:
examples of what these names would be? I am worried about unlimited cardinality which would blow up people's span name drop-down

Reply:
Gotcha ... I wasn't really sure where this was used. Where I work we haven't got a UI around sleuth yet. Currently we just use it to correlate log messages in sumo logic. But some examples could be:

* `Job: DEFAULT.memberIndexingJob; Trigger: DEFAULT.MT_4b7k81ajfts4g`
* `Job: DEFAULT.memberIndexingJob; Trigger: DEFAULT.memberIndexingScheduledTrigger`
* `Job: INDEXING.memberIndexingJob; Trigger: INDEXING.memberIndexingScheduledTrigger`

Maybe just job key alone would be sufficient? I just don't have real world experience where this is used to make that call.

Reply:
Oh...looks like you can add tags to the span? I'm not sure which would be more useful where, but maybe the job key as the name and tag it with the trigger key? Or vice versa?

Reply:
job could possibly be a decent choice. if that also could be unlimited, we sometimes look for a "command name". trigger looks like a bad thing for a name. in this case a tag like `quartz.trigger` if in fact this is something clarifying.

Reply:
I've updated it to use the job key as the name and tagged it with the trigger key. Please let me know if there is something different you'd rather see. I'm completely open to other names/tags.